liquibase-core/src/main/java/liquibase/sqlgenerator/core/AddPrimaryKeyGenerator.java

```diff
@@ -26,9 +26,15 @@ public ValidationErrors validate(AddPrimaryKeyStatement addPrimaryKeyStatement,
         validationErrors.checkRequiredField("columnNames", addPrimaryKeyStatement.getColumnNames());
         validationErrors.checkRequiredField("tableName", addPrimaryKeyStatement.getTableName());
 
-        if (!((database instanceof MSSQLDatabase) || (database instanceof MockDatabase))) {
-            if ((addPrimaryKeyStatement.isClustered() != null) && !addPrimaryKeyStatement.isClustered()) {
-                validationErrors.checkDisallowedField("clustered", addPrimaryKeyStatement.isClustered(), database);
+        if (addPrimaryKeyStatement.isClustered() != null) {
+            if (database instanceof PostgresDatabase) {
+                if (addPrimaryKeyStatement.isClustered() && addPrimaryKeyStatement.getConstraintName() == null) {
+                    validationErrors.addError("Postgresql requires constraintName on addPrimaryKey when clustered=true");
+                }
+            } else if (database instanceof MSSQLDatabase || database instanceof MockDatabase) {
```

Comment:
@nvoxland Should we remove else if for MSSQLDatabase and MockDatabase?

Reply:
No, we want to keep the mssql behavior as-is